CMakeLists.txt

```diff
@@ -30,6 +33,10 @@ else()
 	endif()
 endif()
 
+if(WIN32)
+	string(APPEND CMAKE_CXX_FLAGS " -DNOMINMAX")
```

Comment:
-DNOMINMAX would be better done as a #define inside the Ninja code anywhere windows.h is included.

Reply:
why?

Reply:
Because in the future where parts of ninja can be consumed as a library by other applications, you don't want to infect those applications with the windows.h header file defining a preprocessor macro for min or max, nor do you want to infect those other applications with your library's compiler definitions.

Including windows.h as
```
#define NOMINMAX
#define _WIN32_LEAN_AND_MEAN
#include <windows.h>
#undef NOMINMAX
#undef _WIN32_LEAN_AND_MEAN
```

is the least intrusive way to do it

Reply:
Using target_compile_definitions (with PRIVATE) the definition won't be inherited by other applications who use ninja as a library.

Reply:
That is true. But doesn't address my other concerns. Unless no header files for ninja include windows.h

Reply:
Public header files shouldn't include windows.h at all.